twopaneview/src/TwoPaneView.tsx

```diff
@@ -3,57 +3,54 @@ import {
   View,
   Dimensions,
 } from 'react-native';
-import { DualScreenInfo } from 'react-native-dualscreeninfo'
+import { DualScreenInfo, DualScreenInfoPayload, WindowRect, DeviceRotation } from 'react-native-dualscreeninfo'
 import {Orientation, PanePriority, PaneMode} from "./types"
 
-type Props = {
-} & Partial<DefaultProps>;
-
-type DefaultProps = Readonly<typeof defaultProps>;
-
-const defaultProps = { panePriority: PanePriority.Pane1, 
-  panePriorityVerticalSpanning: Orientation.Horizontal, 
-  paneMode: PaneMode.Auto
-}
-
 type State = {
-  dims: any,
+  windowRects: WindowRect[],
   spanning: boolean, 
-  panePriority?: string,
-  panePriorityVerticalSpanning?: string,
-  paneMode?: string,
-};
+  rotation: DeviceRotation,
+}
+
+export interface TwoPaneViewProps {
+  panePriority: PanePriority,
+  paneMode: PaneMode,
+}
 
-export class TwoPaneView extends Component<Props, State> {
+export class TwoPaneView extends Component<TwoPaneViewProps, State> {
   state: State = {
-    dims: Dimensions.get('window'),
+    windowRects: DualScreenInfo.windowRects,
     spanning: DualScreenInfo.isSpanning, 
-    panePriority: this.props.panePriority,
-    panePriorityVerticalSpanning: this.props.panePriorityVerticalSpanning,
-    paneMode: this.props.paneMode,
+    rotation: DualScreenInfo.rotation,
   };
 
+  getPanePriority() {
+    return this.props.panePriority || PanePriority.Pane1;
+  }
+
+  getPaneMode() {
+    return this.props.paneMode || PaneMode.Auto;    
+  }
+
   componentDidMount() {
-    Dimensions.addEventListener('change', this._handleDimensionsChange);
+    DualScreenInfo.addEventListener('didUpdateSpanning', this._handleSpanningChanged);
   }
 
   componentWillUnmount() {
-    Dimensions.removeEventListener('change', this._handleDimensionsChange);
+    DualScreenInfo.removeEventListener('didUpdateSpanning', this._handleSpanningChanged);
   }
 
-  _handleDimensionsChange = (dimensions: { window: any; }) => {
+  _handleSpanningChanged = (update: DualScreenInfoPayload) => {
     this.setState({
-      dims: dimensions.window,
-      spanning: DualScreenInfo.isSpanning 
+      spanning: update.isSpanning,
+      windowRects: update.windowRects,
+      rotation: update.rotation       
     });
   };
 
   render() {    
-
-    let direction:any ='row';
-
     return (
-      <View style={{flexDirection: direction, width: this.state.dims.width, height:this.state.dims.height}}>
+      <View style={{flexDirection: this.isHorizontalOrientation() ? 'row' : 'column'}}>
```

Comment:
Great,  now with isHorizontalOrientation, orientation detection should be more reliable